src/js/editor-ocr.js

```diff
@@ -0,0 +1,148 @@
+ * @param {int} imageId - Image ID.
+*/
+const insertDescription = async ( imageBlockIndex, imageId ) => {
+	const imageDescription = await getImageDescription( imageId );
+
+	if( !imageDescription ) {
+		return;
+	}
+
+	const newBlock = createBlock( 'core/paragraph', {
+		content: imageDescription,
+		anchor: `classifai-image-description-${imageId}`,
+	} );
+
+	dispatch( 'core/block-editor' ).insertBlock( newBlock, imageBlockIndex + 1 );
+};
+
+/**
+ * An Modal allows user to insert description to block if detected.
+ */
+const imageOcrModal = () => {
+	const [ isOpen, setOpen ] = useState( false );
+	const [ imageId, setImageId ] = useState( 0 );
+	const [ blockIndex, setBlockIndex ] = useState( 0 );
+	const openModal = () => setOpen( true ); // eslint-disable-line require-jsdoc
+	const closeModal = () => setOpen( false ); // eslint-disable-line require-jsdoc
+	let currentBlocks;
+
+	useSelect( debounce( ( select ) => {
+		const { getSelectedBlock, getBlocks, getBlockIndex } = select( 'core/block-editor' );
+		const newBlocks = getBlocks();
+		const prevBlocks = currentBlocks;
+		currentBlocks = newBlocks;
+
+		const currentBlock = getSelectedBlock();
+
+		if ( ! currentBlock || 'core/image' !== currentBlock.name ) {
+			return;
+		}
+
+		if ( ! currentBlock.attributes.id ) {
+			return;
+		}
+
+		const prevBlock = find( prevBlocks, block => block.clientId === currentBlock.clientId );
+
+		if ( ! prevBlock || prevBlock.attributes.id === currentBlock.attributes.id ) {
+			return;
+		}
+
+		setBlockIndex( getBlockIndex( currentBlock.clientId ) );
+		setImageId( currentBlock.attributes.id );
+		openModal();
+	}, 10 ) );
+
+	return isOpen && <Modal title={__( 'ClassifAI detected text in your image' )}>
+		<p>{__( 'Would you like you insert it as a paragraph under you image block?' )}</p>
+		<Flex align='flex-end' justify='flex-end'>
+			<FlexItem>
+				<Button isPrimary onClick={() => {
+					insertDescription( blockIndex, imageId );
+					return closeModal();
+				}}>
+					{__( 'Insert text' )}
+				</Button>
+			</FlexItem>
+			<FlexItem>
+				<Button isSecondary onClick={ closeModal }>
+					{__( 'Dismiss' )}
+				</Button>
+			</FlexItem>
+		</Flex>
+	</Modal>;
+};
+
+registerPlugin( 'tenup-classifai-ocr-modal', {
+	render: imageOcrModal,
+} );
+
+/**
+ * Insert ClassifAI panel to image settings sidebar.
+*/
+const imageOcrControl = createHigherOrderComponent( ( BlockEdit ) => { // eslint-disable-line no-unused-vars
+	return ( props ) => {
+		const { attributes, clientId, isSelected, name } = props;
+
+		if ( ! isSelected || 'core/image' != name ) {
+			return <BlockEdit {...props} />;
+		}
+
+		return (
+			<Fragment>
+				<BlockEdit {...props} />
+				<InspectorControls>
+					<PanelBody title={__( 'ClassifAI' )} initialOpen={true}>
+						<PanelRow>
+							<Button onClick={() => insertDescription( clientId, attributes.id )} isPrimary>
+								{__( 'Insert description' )}
```

Comment:
```suggestion
								{__( 'Insert scanned text into content' )}
```

Maybe more like this? I don't think being stored in the description field or being known as descriptive text is how you think of it in the editorial flow.